src/providers/SocketProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { createContext, useEffect, useReducer, useState } from 'react';
+import { CHAT_URL } from '../config';
+import { IGetMessage, ISocketMessage } from '../interfaces';
+import { addMessage, addRoomID, chatReducer } from '../reducers';
+import { IAction, IChat, IStudent, ITextMessage } from '../interfaces';
+
+export const SocketContext = createContext({
+  uniqueID: '' as string,
+
+  chats: [] as IChat[],
+  dispatchChats(action: IAction) {},
+
+  queue: [] as IStudent[],
+  setQueue(state: IStudent[]) {},
+
+  socketSend(message: ISocketMessage | IGetMessage): void {},
+});
+
+let socket;
+
+const getSocket = (): WebSocket => {
+  if (!socket) socket = new WebSocket(CHAT_URL);
+  return socket;
+};
+
+export const SocketProvider = ({ children }: any) => {
+  const [chats, dispatchChats] = useReducer(chatReducer, []);
+  const [uniqueID, setUniqueID] = useState<string>('');
+  const [queue, setQueue] = useState<IStudent[]>([]);
+
+  const socketHandler = (message): void => {
+    const parsedMessage: ISocketMessage = JSON.parse(message.data);
+
+    if (parsedMessage.type === 'textMessage') {
+      const action = addMessage(
+        {
+          message: parsedMessage.payload['message'],
```

Comment:
Trekk ut payLoad fra parsedMessage for å slippe å skrive parsedMessage flere steder.